controllers/loadtest_controller.go

```diff
@@ -196,12 +212,39 @@ func (r *LoadTestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 	}
 
+	requeueTime := getRequeueTime(test, previousStatus, log)
+	if requeueTime != 0 {
+		return ctrl.Result{RequeueAfter: requeueTime}, nil
+	}
 	return ctrl.Result{}, nil
 }
 
-// LoadTestMissing categorize missing components based on their roles at specific
-// moment. The struct is a wrapper to help us get role information associate
-// with components.
+// getRequeueTime takes a LoadTest and its previous state, and compares the
```

Comment:
I would rewrite this as follows:

```go
// getRequeueTime takes a LoadTest and its previous status, compares the
// previous status of the load test with its updated status, and returns a
// calculated requeue time. If the test has just been assigned a start time
// (i.e., it has just started), the requeue time is set to the timeout value
// specified in the LoadTest. If the test has just been assigned a stop time
// (i.e., it has just terminated), the requeue time is set to the time-to-live
// specified in the LoadTest, minus its actual running time. In other cases,
// the requeue time is set to zero.
```

Reply:
This is great, thank you for reviewing! I forgot to describe what the default value is. I would describe what the function does in all the cases next time, this is great to learn.